docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,244 @@
+
+    # 4. Define dependency.
+    # 4.1 Using a new script - define a new source code.
+    custom_socket.add_dependency(
+        ScriptDependency(
+            name='custom_script'
+            script=Script(
+                runtime_name=RuntimeChoices.PYTHON_V5_0,
+                source='print("custom_script")'
+            )
+        )
+    )
+    # 4.2 Using an existing script.
+    another_custom_script = Script.please.get(id=2)
+    custom_socket.add_dependency(
+        ScriptDependency(
+            name='another_custom_script',
+            script=another_custom_script
+        )
+    )
+
+    # 4.3 Using an existing ScriptEndpoint.
+    script_endpoint = ScriptEndpoint.please.get(name='script_endpoint_name')
+    custom_socket.add_dependency(
+        script_endpoint=script_endpoint
+    )
+
+    # 5. Publish custom_socket.
+    custom_socket.publish()  # this will make an API call and create a script;
+
+Sometimes, it's needed to set up the environment for the custom socket.
+It's possible to check the custom socket status::
+
+    # Reload will refresh object using Syncano API.
+    custom_socket.reload()
+    print(custom_socket.status)
+    # and
+    print(custom_socket.status_info)
+
+Updating the custom socket
+--------------------------
+
+To update custom socket, use::
+
+    custom_socket = CustomSocket.please.get(name='my_custom_socket')
+
+    # to remove endpoint/dependency
+    
+    custom_socket.remove_endpoint(endpoint_name='my_endpoint')
+    custom_socket.remove_dependency(dependency_name='custom_script')
+
+    # or to add a new endpoint/dependency:
+
+    custom_socket.add_endpoint(new_endpoint)  # see above code for endpoint examples;
+    custom_socket.add_dependency(new_dependency)  # see above code for dependency examples;
+
+    # save changes on Syncano
+    
+    custom_socket.update()
+
+
+Running custom socket
+-------------------------
+
+To run a custom socket use::
+
+    # this will run `my_endpoint` - and call `custom_script` (using GET method);
+    result = custom_socket.run(method='GET', endpoint_name='my_endpoint')
+
+
+Read all endpoints in a custom socket
+-----------------------------------
+
+To get the all defined endpoints in a custom socket run::
+
+    endpoints = custom_socket.get_endpoints()
+
+    for endpoint in endpoints:
+        print(endpoint.name)
+        print(endpoint.calls)
+
+To run a particular endpoint::
+
+    endpoint.run(method='GET')
+    # or:
+    endpoint.run(method='POST', data={'name': 'test_name'})
+
+Data will be passed to the API call in the request body.
+
+Read all endpoints
+------------------
+
+To get all endpoints that are defined in all custom sockets::
+
+    socket_endpoint_list = SocketEndpoint.get_all_endpoints()
+
+Above code will return a list with SocketEndpoint objects. To run such endpoint, use::
+
+    socket_endpoint_list.run(method='GET')
+    # or:
+    socket_endpoint_list.run(method='POST', data={'custom_data': 1})
```

Comment:
I also didn't check `Read all endpoints` section as I'm not sure I understand it - so it's possible to run all endpoints from all sockets at once?